lib/partition_setup.pm

```diff
@@ -463,7 +463,7 @@ Select the first hard disk.
 
 =cut
 sub select_first_hard_disk {
-    my @tags           = qw(existing-partitions hard-disk-dev-sdb-selected hard-disk-dev-non-sda-selected select-hard-disks-one-selected);
+    my @tags = qw(existing-partitions hard-disk-dev-sdb-selected hard-disk-dev-sdc-selected hard-disk-dev-sdd-selected hard-disk-dev-non-sda-selected select-hard-disks-one-selected);
```

Comment:
The kernel change LGTM. My only concern is about changing the hard-disk-dev-xxx-selected. It is very easy to lead to  issues during partitioning depending on how needles are added. It needs to be very careful. And different SUTs having different disk topology and initial selected disks may meet problems if not carefully handled. I used to meet this and spent quite some time to solve it on different machines one by one. I tried to open the needle, but could not open it. So I only raise the concern to you and if possible would suggest you to separate this change into a single commit so that if in future it leads to failures, one can quickly revert it while keeping all of your other changes in.

Reply:
@alice-suse this is good idea! I will separate this change. The only place I touched is the `select_first_hard_disk` function - the **first hard disk** should be in my opinion always `sda` so `sdb` shouldn't be selected. But in our case, we have also `sdc` and sometimes `sdd` drives so this change unselects those the same way as it already unselects `sdb`.